### ISSUE TITLE
db.load should follow node 'modules as folders' convention

### ISSUE THREAD

**Issue by nicholasf** [dresende/node-orm2#60]
It's important to support clear separation of model logic and the natural way to do it is via node modules. 

The db.load(..) function is how this problem is solved, as it can load model definitions that are in separate modules. The way it locates them, however, is limited. You have to directly specify an index like file that is in the _same_ directory as the file making the db.load call:

```
orm.connect(env.connectionString, function(err, db){
    db.load("./models/index", function (err) {
        //...
    });
});

```

It would be nice if db.load could work with the traditional loading sequence described here:  http://nodejs.org/api/modules.html#modules_folders_as_modules

In other words, I should be able to do this:

```
orm.connect(env.connectionString, function(err, db){
    db.load("./models/", function (err) {
      //...
    });
});
```

Where models/index.js would be loaded if it exists.
